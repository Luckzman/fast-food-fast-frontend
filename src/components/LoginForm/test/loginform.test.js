import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { Login } from '..';

describe('App', () => {
  const props = {
    login: () => {}
  };
  it('should match snapshot', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should stimulate on submit event', () => {
    const wrapper = shallow(<Login {...props} />);
    wrapper.find('Form').simulate('submit', {
      preventDefault: jest.fn()
    });
    wrapper.update();
    expect(wrapper.find('Form')).toBeDefined();
  });

  it('should change the state after changing the article title', () => {
    const event = { target: { name: 'email' } };
    const wrapper = shallow(<Login />);
    const handleChangeSpy = sinon.spy(wrapper.instance(), 'handleChange');
    wrapper.update();
    wrapper.find('#email').simulate('change', event);
    expect(handleChangeSpy.calledOnce).toBeDefined();
  });
});
