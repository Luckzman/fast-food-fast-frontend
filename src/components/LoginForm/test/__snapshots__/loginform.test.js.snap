// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<Form
  as="form"
  onSubmit={[Function]}
>
  <FormField>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      value=""
    />
  </FormField>
  <FormField>
    <input
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
  </FormField>
  <Button
    as="button"
    onClick={[Function]}
    type="submit"
  >
    Submit
  </Button>
</Form>
`;
